src/Tasks/Microsoft.NET.Build.Tasks/GenerateResxCodeBehindFile.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            switch (char.GetUnicodeCategory(ch))
+            {
+                case UnicodeCategory.UppercaseLetter:
+                case UnicodeCategory.LowercaseLetter:
+                case UnicodeCategory.TitlecaseLetter:
+                case UnicodeCategory.ModifierLetter:
+                case UnicodeCategory.OtherLetter:
+                case UnicodeCategory.LetterNumber:
+                    return true;
+
+                default:
+                    return false;
+            }
+        }
+
+        private void WriteCSharpCodeBehind(ITaskItem resxFile, string codeBehindFile)
+        {
+            StringBuilder builder = new StringBuilder();
+
+            builder.AppendLine("// <auto-generated />");
+            builder.AppendLine("using System.Reflection");
+            builder.AppendLine();
+
+            string resourceName = resxFile.GetMetadata("ManifestResourceName");
+            string[] nameParts = resourceName.Split('.');
+            string namespaceName, className, classIndent;
+            if (nameParts.Length > 1)
+            {
+                namespaceName = string.Join(".", nameParts, 0, nameParts.Length - 1);
+                className = nameParts.Last();
+                classIndent = "    ";
+            }
+            else
+            {
+                namespaceName = null;
+                className = nameParts[0];
+                classIndent = "";
+            }
+
+            string memberIndent = classIndent + "    ";
+
+            if (namespaceName != null)
+                builder.AppendLine($"namespace {namespaceName} {{");
+
+            builder.AppendLine($"{classIndent}internal static class {className}");
+            builder.AppendLine("{");
+
+            builder.AppendLine($"{memberIndent}internal static global::System.Globalization.CultureInfo Culture {{ get; set; }}");
+            builder.AppendLine($"{memberIndent}internal static global::System.Resources.ResourceManager ResourceManager {{ get; }} = new global::System.Resources.ResourceManager(\"{resourceName}\", typeof({className}).GetTypeInfo().Assembly);");
+            builder.AppendLine();
+
+            foreach (var node in XDocument.Load(resxFile.ItemSpec).Descendants("data"))
+            {
+                string name = node.Attribute("name")?.Value ?? throw new InvalidDataException("Missing resource name");
+                string value = node.Elements("value").FirstOrDefault()?.Value.Trim() ?? throw new InvalidDataException($"Missing resource value: '{name}'");
+                string comment = node.Elements("comment").FirstOrDefault()?.Value.Trim();
+
+                if (name == "")
+                    throw new InvalidDataException("Empty resource name");
+
+                // If the comment was provided by the developer, use that.
+                // Otherwise, generate a default comment like the old ResX generator does.
+                if (comment == null)
```

Comment:
In our own usage, comments are  usually notes to translators. I think it would be useful to see both the English text AND the comment in quick info.